service_discovery_manager/Component/mmSH/client_runner.cpp

```diff
@@ -303,14 +303,18 @@ int ClientRunner::Run() {
       meta->monitor_port = info->monitor_port;
 
       meta->client = new MonitorClient(meta->id);
-      meta->message_handle = GetThreadMsgInterface(meta->id);
-      monitor_manager.AddTail(meta);
-
-      CSTI<CbDispatcher>::getInstancePtr()->Subscribe(
-          MONITOR_RESPONSE_EVENT, (void*)meta->message_handle, OnMonitorClientEvent);
-      meta->client->Start(info->address, info->monitor_port);
-      CHAR* monitor_packet = const_cast<CHAR*>("QUERY-MONITORING");
-      meta->client->DataSend(monitor_packet, strlen(monitor_packet) + 1);
+      if (meta->client->Start(info->address, info->monitor_port)) {
+        meta->message_handle = GetThreadMsgInterface(meta->id);
+        CSTI<CbDispatcher>::getInstancePtr()->Subscribe(
+            MONITOR_RESPONSE_EVENT, (void*)meta->message_handle, OnMonitorClientEvent);
+        monitor_manager.AddTail(meta);
+
+        CHAR* monitor_packet = const_cast<CHAR*>("QUERY-MONITORING");
+        meta->client->DataSend(monitor_packet, strlen(monitor_packet) + 1);
+      } else {

```

Comment:
Should `else` exist?
Shouldn't `meta` and `meta->client` always `SAFE_DELETE()`?

Reply:
`meta` and `meta->client` will be deleted after receiving response in `OnMonitorClientEvent` in case of success.
When `MonitorClient::Start` is failed, we should delete them here.